lib/Bagger/Agent/LW.pm

```diff
@@ -76,49 +81,91 @@ _recv() function.
 our %_INJECTION = ();
 
 sub _run_injection {
-    my ($injectpoint, @args) = @_;
-    return $_INJECTION{$injectpoint}(@args) if ref $_INJECTION{$injectpoint};
+    my $injectpoint = shift @_;
+    $_INJECTION{$injectpoint}(@_) if ref $_INJECTION{$injectpoint};
```

Comment:
Re-using the argslist here is not usually considered a best practice, but here it is clear that we are just shifting one off the left side of the array and passing the rest through.  I should probably note this in the coding conventions document proposal